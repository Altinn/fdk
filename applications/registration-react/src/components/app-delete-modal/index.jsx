import React from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

const AppDeleteModal  = (props) => (
  <div>
    <Modal isOpen={props.modal} toggle={() => props.toggle} className={props.className}>
      <ModalHeader toggle={props.toggle}>{props.title}</ModalHeader>
      <ModalBody>
        {props.body}
      </ModalBody>
      <ModalFooter
        className="d-flex justify-content-center"
      >
        <Button color="danger" onClick={() => props.handleAction()}>Ja, slett datasettet</Button>
        <Button color="primary" onClick={props.toggle}>Nei, behold datasettet</Button>
      </ModalFooter>
      <ModalFooter />
    </Modal>
  </div>
);


export default AppDeleteModal;
