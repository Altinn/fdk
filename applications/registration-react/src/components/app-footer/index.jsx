import React from 'react';
import PropTypes from 'prop-types';
//import cx from 'classnames';
import localization from '../../utils/localization';

const Footer  = (props) => {
  return (

      <footer className="fdk-footer">
        <div className="container">
          <div className="row">
            <div className="col-md-3">
              <p className="fdk-p-footer">
                <a
                  href="https://www.brreg.no/personvernerklaering/"
                >
                  {localization.footer.information}<br />
                  {localization.footer.privacy}
                  <i className="fa fa-external-link fdk-fa-right" />
                </a>
              </p>
            </div>
            <div className="col-md-6 text-center">
              <span className="uu-invisible" aria-hidden="false">Felles Datakatalog.</span>
              <p className="fdk-p-footer">
                {localization.footer.information_text}
              </p>
            </div>
            <div className="col-md-3 text-right">
              <p className="fdk-p-footer">
                <a
                  href="mailto:fellesdatakatalog@brreg.no"
                >
                  <span className="uu-invisible" aria-hidden="false">Mailadresse.</span>
                  {localization.footer.contact}<br />
                  {localization.footer.mail}
                </a>
              </p>
            </div>
          </div>
        </div>
      </footer>

  )
}

Footer.defaultProps = {

};

Footer.propTypes = {

};

export default Footer;
