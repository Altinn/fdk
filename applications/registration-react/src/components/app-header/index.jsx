import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';

import {
  fetchUserIfNeeded
} from '../../actions/index';
import localization from '../../utils/localization';
import './index.scss';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
    this.props.dispatch(fetchUserIfNeeded());
  }

  toggle() {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  }

  render() {
    const { userItem } = this.props;
    return (
      <header>
        <div>
          <a
            id="focus-element"
            className="uu-invisible"
            href={`${location.pathname}#content`}
            aria-hidden="true"
          >
            Hopp til hovedinnhold
          </a>
        </div>
        <div id="skip-link-wrap">
          <a id="skip-link" href={`${location.pathname}#content`}>Hopp til hovedinnhold</a>
        </div>
        <div className="fdk-header-beta">
          {localization.beta.header}
          <br className="d-md-none" />
          {localization.beta.first}
          <a className="white-link" href="mailto:fellesdatakatalog@brreg.no">{localization.beta.second}</a> {localization.beta.last}
        </div>
        <div className="fdk-header">
          <div className="container">
            <div className="row">
              <div className="col-6 col-md-4">
                <a
                  title="Link til Felles datakatalog"
                  href="/"
                >
                  <span className="uu-invisible" aria-hidden="false">GÃ¥ til forside</span>
                  <img className="fdk-logo" src="/static/img/fdk-logo@2x.png" alt="Logo for Felles datakatalog" />
                </a>
              </div>

              <div className="col-6 col-md-4 d-flex justify-content-center align-items-center">
                <span><strong>Registrering</strong></span>
              </div>
              <div className="col-md-4 d-flex align-items-center fdk-header-text_items justify-content-end">
                {userItem && userItem.name &&
                  <div className="fdk-margin-right-double">
                    <i className="fa fa-user fdk-fa-left fdk-color-cta3" />
                    {userItem.name}
                  </div>
                }
                <div className="fdk-margin-right-double fdk-auth-link">
                  <a href={`${window.location.origin  }/logout`}>{localization.app.logOut}</a>
                </div>
                <div>
                  <ButtonDropdown className="btn-group-default" isOpen={this.state.dropdownOpen} toggle={this.toggle}>
                    <DropdownToggle className="fdk-button fdk-button-default fdk-button-menu dropdown-toggle btn-default" caret>
                      <i className="fa fa-bars fdk-fa-dark fdk-fa-left" />
                      <span>Meny</span>
                    </DropdownToggle>
                    <DropdownMenu>
                      <DropdownItem>Veileder</DropdownItem>
                      <DropdownItem>Standard</DropdownItem>
                      <DropdownItem>
                        <a href="">Felles datakatalog</a>
                      </DropdownItem>
                    </DropdownMenu>
                  </ButtonDropdown>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="fdk-breadcrumb breadcrumb d-flex justify-content-between">
          <span></span>
          <span>{localization.app.autoSave}</span>
        </div>
      </header>
    );
  }
}

Header.defaultProps = {

};

Header.propTypes = {
  dispatch: PropTypes.func.isRequired
};

function mapStateToProps({ user }, ownProps) {
  const { userItem } = user || {
    userItem: null
  }

  return {
    userItem
  };
}

export default connect(mapStateToProps)(Header);
