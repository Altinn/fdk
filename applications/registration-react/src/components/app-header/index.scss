@import '../../assets/style/colors';

.fdk-header-text_items {
  font-size: 0.9rem;
  white-space: nowrap;
}

.fdk-auth-link {
  a {
    color: $color1;
    font-size: .9rem;
  }
}