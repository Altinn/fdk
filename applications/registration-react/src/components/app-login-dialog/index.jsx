import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import axios from 'axios';

import localization from '../../utils/localization';
import './index.scss';

//const LoginDialog = props => {
class LoginDialog extends React.Component {
  componentWillMount() {
    if(this.props.loggedOut) {
      return axios.get('/logout')
        .then((response) => response)
        .catch((response) => {
          const { error } = response;
          return Promise.reject(error);
        })

      return axios.get(
        '/logout'
      ).then((response) => {
      }).catch((error) => {
        throw {error}
      });
    }
  }
  render() {
    const {loggedOut} = this.props;
    return (
      <div className="login-dialog-wrapper p-5">
        <div className="container-fluid">
          <div className="row">
            <div className="col-12 col-md-8 offset-md-2">
              <div className="col-12 col-md-8">
                {loggedOut &&
                (
                  <div className="mt-2 alert alert-warning fdk-text-size-small fdk-color1" role="alert">
                    <span>
                      <strong>{localization.loginDialog.loggedOutMsgPart1}</strong>
                    </span>
                    <span>
                      {localization.loginDialog.loggedOutMsgPart2}
                    </span>
                  </div>
                )
                }
                {!loggedOut &&
                (
                  <div>
                    <h1 className="fdk-text-extra-strong mb-md-5">{localization.loginDialog.title}</h1>
                    <div className="fdk-text-size-medium fdk-text-line-medium mt-2 mb-md-3">
                      {localization.loginDialog.ingress}
                    </div>
                  </div>
                )
                }
                <a className="fdk-button fdk-button-cta mb-2 mb-md-5" href="/login">
                  {localization.app.logIn}
                </a>
                <div className="fdk-text-size-small fdk-text-line-medium">
                  <strong>{localization.catalogs.missingCatalogs.accessTitle}</strong>
                  <p>
                    {localization.catalogs.missingCatalogs.accessText}
                  </p>
                  <strong>{localization.catalogs.missingCatalogs.assignAccessTitle}</strong>
                  <p>
                    {localization.catalogs.missingCatalogs.assignAccessText}
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default LoginDialog;
