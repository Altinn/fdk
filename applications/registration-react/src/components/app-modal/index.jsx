import React from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

const AppModal  = (props) => (
  <div>
    <Modal isOpen={props.modal} toggle={() => props.toggle} className={props.className}>
      <ModalHeader toggle={props.toggle}>{props.title}</ModalHeader>
      <ModalBody>
        {props.body}
      </ModalBody>
      <ModalFooter>
        <Button color="secondary" onClick={props.toggle}>Cancel</Button>
      </ModalFooter>
    </Modal>
  </div>
);


export default AppModal;
