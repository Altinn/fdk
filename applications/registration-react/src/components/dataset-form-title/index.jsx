// {/*
// import React from 'react';
// import { withFormik, Form, Field } from 'formik';
// import Yup from 'yup';
//
// const DatasetForm2 = ({
//   values,
//   errors,
//   touched,
//   setTouched
// }) => (
//   <div>
//     { touched.title && errors.title && <p>{errors.title}</p> }
//     <Field
//       type="text"
//       name="title"
//       placeholder="Title"
//       value={values.title}
//       onBlur={(e) => {values.onSave(e, setTouched, values, errors)}}
//     />
//   </div>
// );
//
// export class DatasetForm3 extends React.Component { // eslint-disable-line react/prefer-stateless-function
//   constructor(props) {
//     super(props);
//     this.state = ({
//       title: this.props.values.title
//     })
//   }
//
//   componentWillReceiveProps() {
//
//   }
//
//   render() {
//     const { values, errors, touched, setTouched } = this.props;
//     return (
//       <Form refs="myForm">
//         { touched.title && errors.title && <p>{errors.title}</p> }
//         <Field type="text" name="title" placeholder="Title" value={values.title} onBlur={(e) => {
//           values.onSave(e, setTouched, values, errors)
//         }}/>
//         <Field type="text" name="name" placeholder="Name" onBlur={(e) => {
//           handleBlur(e, setTouched, values, errors)
//         }}/>
//         <button>Submit</button>
//       </Form>
//     );
//   }
//
// }
//
// const DatasetForm  = (props) => {
//   const { values, errors, touched, setTouched } = props;
//
//   return (
//     <Form refs="myForm">
//       {DatasetForm2({values, errors, touched, setTouched})}
//       {DatasetForm4({values, errors, touched, setTouched})}
//     </Form>
//   );
//
// }
//
// const DatasetForm4 = ({
//   values,
//   errors,
//   touched,
//   setTouched
// }) => (
//   <div>
//     <Field type="text" name="name" placeholder="Name" onBlur={(e) => {
//       values.onSave(e, setTouched, values, errors)
//     }}/>
//     <button>Submit</button>
//   </div>
// );
//
// function handleBlur(e, setTouched, values, errors) {
//
//   /*
//   console.log('values', values);
//   console.log('errors', errors);
//   setTouched({[e.target.name]: true});
//   const saveValue = e.target.value
//   console.log("handle custom change", e.target.name + ' ' + saveValue)
//
//
//   const body = {
//     id: "e679b150-e69d-444c-bf7f-874d6999c62d",
//     uri: "http://brreg.no/catalogs/910244132/datasets/e679b150-e69d-444c-bf7f-874d6999c62d",
//     title: {
//       nb: `${saveValue}`
//     },
//     description: null,
//     objective: null,
//     publisher: {
//       uri: "http://data.brreg.no/enhetsregisteret/enhet/910244132",
//       id: "910244132",
//       name: "RAMSUND OG ROGNAN REVISJON"
//     },
//     accessRights: {
//       uri: "http://publications.europa.eu/resource/authority/access-right/RESTRICTED"
//     },
//     catalogId: "910244132",
//     _lastModified: "2018-01-12T10:58:07.524+0000",
//     registrationStatus: "DRAFT",
//     keyword: [{
//       nb: "SÃ¸keord 3"
//     }],
//     subject: null,
//     theme: null,
//     landingPage: null,
//     identifier: null,
//     contactPoint: null,
//     conformsTo: null,
//     distribution: null,
//     sample: null,
//     language: null,
//     temporal: null,
//     legalBasisForProcessing: null,
//     legalBasisForAccess: null,
//     informationModel: null,
//     references: null,
//     issued: null,
//     modified: null,
//     type: null,
//     accrualPeriodicity: null,
//     provenance: null,
//     hasCurrentnessAnnotation: null,
//     spatial: [{
//       uri: "hurum",
//       prefLabel: null
//     }],
//     legalBasisForRestriction: [{
//       uri: null,
//       prefLabel: {
//         nb: "testerPrefLabel"
//       }
//     }]
//   }
//   const api = {
//     Authorization: "Basic " + null
//   }
//
//   axios.put(
//     '/catalogs/910244132/datasets/e679b150-e69d-444c-bf7f-874d6999c62d/',body, {headers: api}
//   )
//     .then((response) => {
//       console.log("saved!");
//
//       //console.log(JSON.stringify(response));
//     })
//     .catch((error) => {
//       console.log('feiler');
//     })
//   ;
//
//   /*
//    this.delay(() => {
//    if (this.datasetSavingEnabled) {
//    that.save.call(that);
//    }
//    }, this.saveDelay);
//    */
//
//   /*
//    saveDelay: number = 500;
//    */
//
// }
//
// /*
//  function delay(callback, ms): void {
//  clearTimeout(this.timer);
//  this.timer = setTimeout(callback, ms);
//  }
//  */
//
// export default withFormik({
//   mapPropsToValues({ title, onSave, onChange }) {
//     return {
//       title: title || '',
//       onSave
//     }
//   },
//   validationSchema: Yup.object().shape({
//     title: Yup.string().min(3, 'Mininum 3 characters or longer').required('Title is required')
//   }),
//   handleSubmit(values) {
//     console.log(values)
//   }
// })(DatasetForm)
// */}
