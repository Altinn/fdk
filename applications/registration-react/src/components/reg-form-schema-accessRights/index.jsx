import React from 'react';
import PropTypes from 'prop-types';
import { Field, FieldArray, reduxForm, formValueSelector, getFormSyncErrors } from 'redux-form';
import { connect } from 'react-redux';
import _throttle from 'lodash/throttle';

import localization from '../../utils/localization';
import Helptext from '../reg-form-helptext';
import InputField from '../reg-form-field-input';
import RadioField from '../reg-form-field-radio';
import asyncValidate from '../../utils/asyncValidate';
import { accessRights, legalBasisType } from '../../schemaTypes';
import { validateRequired, validateMinTwoChars, validateURL} from '../../validation/validation';

const validate = values => {
  let errors = {}
  const accessRight = (values.accessRights && values.accessRights.uri) ? values.accessRights.uri : null;
  const { legalBasisForRestriction, legalBasisForProcessing, legalBasisForAccess } = values;
  let legalBasisForRestrictionNodes = null;
  let legalBasisForProcessingNodes = null;
  let legalBasisForAccessNodes = null;

  errors = validateRequired('accessRight', accessRight, errors);

  if (legalBasisForRestriction) {
    legalBasisForRestrictionNodes = legalBasisForRestriction.map(item => {
      let itemErrors = {};
      const legalBasisForRestrictionPrefLabel = (item.prefLabel && item.prefLabel.nb) ? item.prefLabel.nb : null;
      const legalBasisForRestrictionURI = item.uri ? item.uri : null;

      itemErrors = validateMinTwoChars('prefLabel', legalBasisForRestrictionPrefLabel, itemErrors);
      itemErrors = validateURL('uri', legalBasisForRestrictionURI, itemErrors);

      return itemErrors;
    });
    let showSyncError = false;
    showSyncError = (legalBasisForRestrictionNodes.filter(item => (item && JSON.stringify(item) !== '{}')).length > 0);
    if (showSyncError) {
      errors.legalBasisForRestriction = legalBasisForRestrictionNodes;
    }
  }

  if (legalBasisForProcessing) {
    legalBasisForProcessingNodes = legalBasisForProcessing.map(item => {
      let itemErrors = {}
      const legalBasisForProcessingPrefLabel = (item.prefLabel && item.prefLabel.nb) ? item.prefLabel.nb : null;
      const legalBasisForProcessingURI = item.uri ? item.uri : null;

      itemErrors = validateMinTwoChars('prefLabel', legalBasisForProcessingPrefLabel, itemErrors);
      itemErrors = validateURL('uri', legalBasisForProcessingURI, itemErrors);

      return itemErrors;
    });
    let showSyncError = false;
    showSyncError = (legalBasisForProcessingNodes.filter(item => (item && JSON.stringify(item) !== '{}')).length > 0);
    if (showSyncError) {
      errors.legalBasisForProcessing = legalBasisForProcessingNodes;
    }
  }

  if (legalBasisForAccess) {
    legalBasisForAccessNodes = legalBasisForAccess.map(item => {
      let itemErrors = {}
      const legalBasisForAccessPrefLabel = (item.prefLabel && item.prefLabel.nb) ? item.prefLabel.nb : null;
      const legalBasisForAccessURI = item.uri ? item.uri : null;

      itemErrors = validateMinTwoChars('prefLabel', legalBasisForAccessPrefLabel, itemErrors);
      itemErrors = validateURL('uri', legalBasisForAccessURI, itemErrors);

      return itemErrors;
    });
    let showSyncError = false;
    showSyncError = (legalBasisForAccessNodes.filter(item => (item && JSON.stringify(item) !== '{}')).length > 0);
    if (showSyncError) {
      errors.legalBasisForAccess = legalBasisForAccessNodes;
    }
  }

  return errors
}

/*
 Resets fields when radio button "Offentlig" is chosen.
 */
const resetFields = (props) => {
  props.change('legalBasisForRestriction', [legalBasisType]);
  props.change('legalBasisForProcessing', [legalBasisType]);
  props.change('legalBasisForAccess', [legalBasisType]);
}

const renderLegalBasisFields = (item, index, fields, customProps) => (
  <div className="d-flex mb-2" key={index}>
    <div className="w-50">
      <Field
        name={`${item}.prefLabel.nb`}
        component={InputField}
        label={customProps.titleLabel}
        showLabel
      />
    </div>
    <div className="w-50">
      <Field
        name={`${item}.uri`}
        component={InputField}
        label={customProps.linkLabel}
        showLabel
      />
    </div>
    <div className="d-flex align-items-end">
      <button
        className="fdk-btn-no-border"
        type="button"
        title="Remove temporal"
        onClick={
          () => {
            if (fields.length === 1) {
              fields.remove(index);
              fields.push({});
            }

            if (fields.length > 1) {
              fields.remove(index);
            }
            asyncValidate(fields.getAll(), customProps.dispatch, customProps, `remove_${fields.name}_${index}`);
          }
        }
      >
        <i className="fa fa-trash mr-2" />
      </button>
    </div>
  </div>
);

const renderLegalBasis = (customProps) => {
  const { fields } = customProps;
  return (
    <div>
      {fields && fields.map((item, index) =>
        renderLegalBasisFields(item, index, fields, customProps)
      )}
      <button className="fdk-btn-no-border" type="button" onClick={() => fields.push({})}>
        <i className="fa fa-plus mr-2" />
        Legg til
      </button>
    </div>
  );
};

export const FormAccessRights = (props) => {
  const { syncErrors, helptextItems, hasAccessRightsURI } = props;
  let accessRight = null;
  if (syncErrors) {
    accessRight = syncErrors.accessRight;
  }
  return (
    <form>
      <div className="form-group">
        <Helptext title="TilgangsnivÃ¥" helptextItems={helptextItems.Dataset_distribution} />
        <Field
          name="accessRights.uri"
          radioId="accessRight-public"
          component={RadioField}
          type="radio"
          value="http://publications.europa.eu/resource/authority/access-right/PUBLIC"
          label="Offentlig"
          onChange={() => resetFields(props)}
        />
        <Field
          name="accessRights.uri"
          radioId="accessRight-restricted"
          component={RadioField}
          type="radio"
          value="http://publications.europa.eu/resource/authority/access-right/RESTRICTED"
          label="Begrenset offentlighet"
        />
        <Field
          name="accessRights.uri"
          radioId="accessRight-non_public"
          component={RadioField}
          type="radio"
          value="http://publications.europa.eu/resource/authority/access-right/NON_PUBLIC"
          label="Unntatt offentlighet"
        />

        {accessRight &&
        <div className="alert alert-danger mt-3">{accessRight.nb}</div>
        }

        {
          (hasAccessRightsURI === 'http://publications.europa.eu/resource/authority/access-right/RESTRICTED' ||
            hasAccessRightsURI === 'http://publications.europa.eu/resource/authority/access-right/NON_PUBLIC'
          ) && (
            <div className="mt-4">
              <div className="form-group">
                <Helptext title={localization.schema.accessRights.legalBasisForRestriction.heading} helptextItems={helptextItems.Dataset_legalBasisForRestriction} />
                <FieldArray
                  name="legalBasisForRestriction"
                  component={renderLegalBasis}
                  titleLabel={localization.schema.accessRights.legalBasisForRestriction.titleLabel}
                  linkLabel={localization.schema.accessRights.legalBasisForRestriction.linkLabel}
                />
              </div>

              <div className="form-group">
                <Helptext title={localization.schema.accessRights.legalBasisForProcessing.heading} helptextItems={helptextItems.Dataset_legalBasisForProcessing} />
                <FieldArray
                  name="legalBasisForProcessing"
                  component={renderLegalBasis}
                  titleLabel={localization.schema.accessRights.legalBasisForProcessing.titleLabel}
                  linkLabel={localization.schema.accessRights.legalBasisForProcessing.linkLabel}
                />
              </div>

              <div className="form-group">
                <Helptext title={localization.schema.accessRights.legalBasisForAccess.heading} helptextItems={helptextItems.Dataset_legalBasisForAccess} />
                <FieldArray
                  name="legalBasisForAccess"
                  component={renderLegalBasis}
                  titleLabel={localization.schema.accessRights.legalBasisForAccess.titleLabel}
                  linkLabel={localization.schema.accessRights.legalBasisForAccess.linkLabel}
                />
              </div>
            </div>
          )}
      </div>
    </form>
  )
}

FormAccessRights.defaultProps = {
  syncErrors: null,
  hasAccessRightsURI: null
};

FormAccessRights.propTypes = {
  syncErrors: PropTypes.object,
  helptextItems: PropTypes.object.isRequired,
  hasAccessRightsURI: PropTypes.string
};

const selector = formValueSelector('accessRights');

let FormAccessRightsSchema = reduxForm({
  form: 'accessRights',
  validate,
  asyncValidate: _throttle(asyncValidate, 250),
})(connect(state => {
  const hasAccessRightsURI = selector(state, 'accessRights.uri')
  return {
    hasAccessRightsURI,
    syncErrors: getFormSyncErrors("accessRights")(state)
  }
})(FormAccessRights));


const mapStateToProps = ({ dataset }) => (
  {
    initialValues: {
      accessRights: (dataset.result.accessRights) ? dataset.result.accessRights : accessRights,
      legalBasisForRestriction: (dataset.result.legalBasisForRestriction && dataset.result.legalBasisForRestriction.length > 0) ? dataset.result.legalBasisForRestriction : [legalBasisType],
      legalBasisForProcessing: (dataset.result.legalBasisForProcessing && dataset.result.legalBasisForProcessing.length > 0) ? dataset.result.legalBasisForProcessing : [legalBasisType],
      legalBasisForAccess: (dataset.result.legalBasisForAccess && dataset.result.legalBasisForAccess.length > 0) ? dataset.result.legalBasisForAccess : [legalBasisType]
    }
  }
)

export default connect(mapStateToProps)(FormAccessRightsSchema)
