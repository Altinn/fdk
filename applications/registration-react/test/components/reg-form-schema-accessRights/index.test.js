import React from 'react';
import { shallow } from 'enzyme';
import { FormAccessRights } from '../../../src/components/reg-form-schema-accessRights';
import helptext from '../../fixtures/helptext';

let defaultProps, wrapper;

beforeEach(() => {
  const { helptextItems } = helptext;
  defaultProps = {
    syncErrors: null,
    helptextItems,
    hasAccessRightsURI: 'hasAccessRightsURI'
  };
  wrapper = shallow(<FormAccessRights {...defaultProps} />);
});

test('should render FormAccessRights correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
