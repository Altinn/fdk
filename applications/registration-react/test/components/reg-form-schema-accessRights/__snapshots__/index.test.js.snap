// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormAccessRights correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <Helptext
      helptextItems={
        Object {
          "description": Object {
            "nb": "For å angi hvor man kan få tilgang til datasettet skal det angis ulike distribusjoner.
* Det angis i utgangspunktet en distribusjon per fil, feed eller API
* Dersom det er ett API som leverer flere filformater angis det som en distribusjon",
          },
          "id": "Dataset_distribution",
          "shortdesc": Object {
            "nb": "For å angi hvor man kan få tilgang til datasettet skal det angis ulike distribusjoner.",
          },
          "uri": "http://brreg.no/fdk/fields#Dataset_distribution",
        }
      }
      required={false}
      title="Tilgangsnivå"
    />
    <Field
      component={[Function]}
      label="Offentlig"
      name="accessRights.uri"
      onChange={[Function]}
      radioId="accessRight-public"
      type="radio"
      value="http://publications.europa.eu/resource/authority/access-right/PUBLIC"
    />
    <Field
      component={[Function]}
      label="Begrenset offentlighet"
      name="accessRights.uri"
      radioId="accessRight-restricted"
      type="radio"
      value="http://publications.europa.eu/resource/authority/access-right/RESTRICTED"
    />
    <Field
      component={[Function]}
      label="Unntatt offentlighet"
      name="accessRights.uri"
      radioId="accessRight-non_public"
      type="radio"
      value="http://publications.europa.eu/resource/authority/access-right/NON_PUBLIC"
    />
  </div>
</form>
`;
